### PR TITLE
Optimize loop.

### PR DESCRIPTION
**Pull request by johnou**

### CODE REVIEW

**Comment by gabrie-allaigre** [comment]
Hi,
Why ? Markdown increments automatically. https://docs.gitlab.com/ee/user/markdown.html#lists

**Comment by johnou** [comment]
@gabrie-allaigre tricky, I'll at least revert that, add a comment and keep the loop optimization.

**Comment by johnou** [comment]
@gabrie-allaigre PTAL

File: src/main/java/com/talanlabs/sonar/plugins/gitlab/GlobalReport.java [hunk 1]
```diff
@@ -109,27 +109,29 @@ private void appendExtraIssues(StringBuilder builder, boolean hasInlineIssues, b
...
         for (Severity severity : SEVERITIES) {
             List<IssueUrl> issueUrls = notReportedOnDiffMap.get(severity);
             if (issueUrls != null && !issueUrls.isEmpty()) {
                 for (IssueUrl issueUrl : issueUrls) {
                     PostJobIssue postJobIssue = issueUrl.postJobIssue;
-                    String msg = "1. " + markDownUtils.globalIssue(postJobIssue.severity(), postJobIssue.message(), postJobIssue.ruleKey().toString(), issueUrl.gitLabUrl, postJobIssue.componentKey());
-                    if (i < gitLabPluginConfiguration.maxGlobalIssues()) {
-                        builder.append(msg).append("\n");
+                    if (reportedIssueCount < gitLabPluginConfiguration.maxGlobalIssues()) {
+                        builder.append(reportedIssueCount)
```
Comment (johnou) [inline src/main/java/com/talanlabs/sonar/plugins/gitlab/GlobalReport.java]: moved the markdownutil call into the if block to only create strings when necessary.

**Comment by gabrie-allaigre** [comment]
Attention, test failed. I will add travis ci.

**Comment by johnou** [comment]
@gabrie-allaigre I'll run it locally and update pull request.

**Comment by johnou** [comment]
@gabrie-allaigre PTAL

**Comment by johnou** [comment]
@gabrie-allaigre PTAL

```
Results :




Tests run: 57, Failures: 0, Errors: 0, Skipped: 0

[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:report (post-unit-test) @ sonar-gitlab-plugin ---
[INFO] Analyzed bundle 'SonarQube :: GitLab Plugin' with 13 classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 6.297 s
[INFO] Finished at: 2017-03-07T12:32:49+01:00
[INFO] Final Memory: 36M/596M
[INFO] ------------------------------------------------------------------------
```

**Comment by johnou** [comment]
Rebased for Travis.

**Comment by johnou** [comment]
@gabrie-allaigre any changes you want before it's pulled in?

**Comment by gabrie-allaigre** [comment]
I integrated the modification into the master directly. Because I rewrite the entire comment section.

**Pull request closed without merging**